rust/flatbuffers/src/builder.rs

```diff
@@ -148,10 +150,9 @@ impl<'fbb> FlatBufferBuilder<'fbb> {
     #[inline]
     pub fn push_slot<X: Push + PartialEq>(&mut self, slotoff: VOffsetT, x: X, default: X) {
         self.assert_nested("push_slot");
-        if x == default {
-            return;
+        if self.force_defaults || x != default {
```

Comment:
turn these two conditions around: since the first is usually false, and the second is usually true, that will be quicker.

Reply:
Note that the first condition is always cheap (reading a bool that should already be in L1 cache) and will usually be branch-predicted correctly, while the second one may be arbitrarily expensive (for example, a string comparison) and less predictable.  Reversing the order may be a tiny amount faster in some cases, but a large amount slower in others.

But I suspect in the common case there is no measurable difference, so I'm happy to make this change.

Reply:
I'm aware its a faster test.. but that doesn't help if its always `false`. We want to optimize for this flag being off, as the "correct" way to use FlatBuffers. @rw